pkg/csi-handler/pv_checker.go

```diff
@@ -203,6 +219,46 @@ func (checker *PVHealthConditionChecker) CheckNodeVolumeStatus(kubeletRootPath s
 
 	if volumeCondition.GetAbnormal() {
 		checker.eventRecorder.Event(pod, v1.EventTypeWarning, "VolumeConditionAbnormal", volumeCondition.GetMessage())
+	} else {
+		return checker.sendRecoveryEventToPod(pod, volumeCondition)
+	}
+
+	return nil
+}
+
+// sendRecoveryEventToPVC sends the recovery event to the pvc
+// If the volume condition is normal and abnormal event wasn't expired,
+// PVHealthConditionChecker should send recovery event.
+func (checker *PVHealthConditionChecker) sendRecoveryEventToPVC(pvc *v1.PersistentVolumeClaim, volumeCondition *VolumeConditionResult) error {
+	pvcUID := string(pvc.ObjectMeta.GetUID())
+	key := fmt.Sprintf("%s:%s:%s", pvcUID, v1.EventTypeWarning, "VolumeConditionAbnormal")
+	events, err := checker.eventInformer.Informer().GetIndexer().ByIndex(util.DefaultEventIndexerName, key)
```

Comment:
Just curious: will expired events be removed from event local cache right now after it is expired ?

Reply:
I didn't check the implementation details. From my understanding, after one period of `event-ttl` passed, kube-apiserver will remove those events within `event-ttl`. In that case, it should trigger `DELETE Event` and handler of deletion should also be executed. So, the expired event in cache also should be removed.